arangod/Aql/IResearchViewOptimizerRules.cpp

```diff
@@ -263,35 +263,88 @@ bool optimizeSort(IResearchViewNode& viewNode, ExecutionPlan* plan) {
   }
 }
 
+bool attributesMatch(IResearchViewSort const& primarySort, latematerialized::NodeWithAttrs& node) {
+  // check all node attributes to be in sort
+  for (auto& nodeAttr : node.attrs) {
+    size_t sortFieldNum = 0;
+    for (auto const& field : primarySort.fields()) {
+      if (arangodb::basics::AttributeName::isIdentical(nodeAttr.attr, field, false)) {
+        nodeAttr.afData.number = sortFieldNum;
+        nodeAttr.afData.field = &field;
+        std::vector<arangodb::basics::AttributeName> empty;
+        nodeAttr.attr.swap(empty); // we do not need later
```

Comment:
why not just `nodeAttr.attr.clear()`?

Reply:
Clear() leaves the capacity() of the vector unchanged.

Reply:
OK: clear() + shrink_to_fit()